### ISSUE TITLE
`@deprecated` w/o argument is deprecated

### ISSUE THREAD

**Issue by daddykotex** [disneystreaming/smithy4s#948]
We generate `@deprecated` annotation code from shapes that w/ the `smithy.api#deprecated` trait. `smithy.api#deprecated` is defined as:

```
/// Marks a shape or member as deprecated.
@trait
structure deprecated {
    /// The reason for deprecation.
    message: String,

    /// A description of when the shape was deprecated (e.g., a date or version).
    since: String,
}
```

So there are fields that we can use to render a proper `@deprecated` annotation in Scala code. I suggested that we get the `message` and `since` from the hint and fallback to `N/A` if empty.

**Comment by daddykotex** [comment]
Note that this can be worked around if your compiler options are strict and you're getting error because of this with the following option: `"""-Wconf:msg=Specify both message and version: @deprecated("message", since = "1.0")
:silent"""`

**Comment by i10416** [comment]
I think rendering deprecated `message` and `since` are already supported here https://github.com/disneystreaming/smithy4s/pull/599, aren't they?

Or do you mean we should render deprecation without message and since as `@deprecated(message = "N/A", since = "N/A")` instead of `@deprecated` to avoid compiler error/warning?

**Comment by kubukoz** [comment]
I think it's the latter, yeah. Basically if either `message` or `since` is missing we'd replace it with `N/A` or an empty string (if that works from the scalac perspective)

**Comment by daddykotex** [comment]
released in 0.17.8
